craft_parts/plugins/cmake_plugin.py

```diff
@@ -0,0 +1,125 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2020-2022 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The cmake plugin."""
+
+from typing import Any, Dict, List, Set, cast
+
+from .base import Plugin, PluginModel, extract_plugin_properties
+from .properties import PluginProperties
+
+
+class CMakePluginProperties(PluginProperties, PluginModel):
+    """The part properties used by the cmake plugin."""
+
+    cmake_parameters: List[str] = []
+    cmake_generator: str = "Unix Makefiles"
+
+    # part properties required by the plugin
+    source: str
+
+    @classmethod
+    def unmarshal(cls, data: Dict[str, Any]):
+        """Populate class attributes from the part specification.
+
+        :param data: A dictionary containing part properties.
+
+        :return: The populated plugin properties data object.
+
+        :raise pydantic.ValidationError: If validation fails.
+        """
+        plugin_data = extract_plugin_properties(
+            data, plugin_name="cmake", required=["source"]
+        )
+        return cls(**plugin_data)
+
+
+class CMakePlugin(Plugin):
+    """The cmake plugin is useful for building cmake based parts.
+
+    These are projects that have a CMakeLists.txt that drives the build.
+    The plugin requires a CMakeLists.txt in the root of the source tree.
+
+    This plugin uses the common plugin keywords as well as those for "sources".
+    For more information check the 'plugins' topic for the former and the
+    'sources' topic for the latter.
+
+    Additionally, this plugin uses the following plugin-specific keywords:
+
+        - cmake-parameters
+          (list of strings)
+          parameters to pass to the build using the common cmake semantics.
+
+        - cmake-generator
+          (string; default: "Unix Makefiles")
+          Determine what native build system is to be used.
+          Can be either `Ninja` or `Unix Makefiles` (default).
+
+    This plugin also interprets these specific build-environment entries:
+
+        - SNAPCRAFT_CMAKE_ARGS
```

Comment:
I would remove this fow now, we used this from the extension for kde but it shouldbe solved some other way

Reply:
we can re-add later if creativity does not allow for us to get there :-)